src/extractor/edge_based_graph_factory.cpp

```diff
@@ -368,181 +377,279 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
         util::Percent progress(log, m_node_based_graph->GetNumberOfNodes());
         // going over all nodes (which form the center of an intersection), we compute all
         // possible turns along these intersections.
-        for (const auto node_at_center_of_intersection :
-             util::irange(0u, m_node_based_graph->GetNumberOfNodes()))
+        std::mutex merge_mutex; // To control merging of all the data into the main data structures
+        std::uint64_t nodes_completed = 0;
+
+        const NodeID node_count = m_node_based_graph->GetNumberOfNodes();
+        NodeID current_node = 0;
+
+        tbb::filter_t<void, NodeID> intersection_ids(tbb::filter::serial_in_order,
```

Comment:
Good call.  I implemented this and it brings processing time down to 25 seconds for the `mexico-latest.osm.pbf` extract I've been testing - basically the same as the non-ordered original approach, but less buggy :-)